src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -0,0 +1,175 @@
+                    arity: 0,
+                    ConsList<TypeSymbol>.Empty,
+                    LookupOptions.MustBeInstance | LookupOptions.MustBeInvocableIfMember,
+                    this,
+                    diagnose: false,
+                    ref useSiteDiagnostics);
+
+                if (lookupResult.IsMultiViable)
+                {
+                    foreach (var symbol in lookupResult.Symbols)
+                    {
+                        if (symbol is MethodSymbol { ParameterCount: 0 } m)
+                        {
+                            cloneMethod = m;
+                            break;
+                        }
+                    }
+                }
+
+                lookupResult.Clear();
+                // PROTOTYPE: discarding use-site diagnostics
+                useSiteDiagnostics = null;
+
+                if (cloneMethod is null)
+                {
+                    hasErrors = true;
+                    diagnostics.Add(ErrorCode.ERR_NoSingleCloneMethod, syntax.Receiver.Location, receiverType);
+                }
+                else
+                {
+                    // Check return type
+                    if (!receiverType.IsEqualToOrDerivedFrom(
+                            cloneMethod.ReturnType,
+                            TypeCompareKind.ConsiderEverything,
+                            ref useSiteDiagnostics))
+                    {
+                        hasErrors = true;
+                        diagnostics.Add(
+                            ErrorCode.ERR_ContainingTypeMustDeriveFromWithReturnType,
+                            syntax.Receiver.Location,
+                            receiverType,
+                            cloneMethod.ReturnType);
+                    }
+
+                    // PROTOTYPE: discarding use-site diagnostics
+                    useSiteDiagnostics = null;
+                }
+            }
+
+            var cloneReturnType = cloneMethod?.ReturnType;
+
+            var args = ArrayBuilder<(Symbol?, BoundExpression)>.GetInstance();
+            // Bind with expression arguments
+            foreach (var initializer in syntax.Initializers)
+            {
+                var propName = initializer.NameEquals?.Name.Identifier.Text;
+                Symbol? member = null;
+                if (!(propName is null) && !(cloneReturnType is null))
+                {
+                    var location = initializer.NameEquals!.Name.Location;
+                    this.LookupMembersInType(
+                        lookupResult,
+                        cloneReturnType,
+                        propName,
+                        arity: 0,
+                        basesBeingResolved: null,
+                        options: LookupOptions.Default,
+                        originalBinder: this,
+                        diagnose: false,
+                        useSiteDiagnostics: ref useSiteDiagnostics);
+                    if (lookupResult.IsSingleViable &&
```

Comment:
>IsSingleViable [](start = 37, length = 14)

How does this handle properties that are overridden or hidden (more than one symbol in the lookup result)?  This does not correspond to how expressions are bound.